### PR TITLE
Add `timm` models to model library

### PR DESCRIPTION
**Pull request by danielholanda**
Add 889 TIMM models to mlagility

15 (<2%) models were skipped for various reasons

### CODE REVIEW

File: models/timm/beit_base_patch16_224.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# Parsing command-line arguments
+batch_size = parse(["batch_size"])
+
+# Creating model and set it to evaluation mode
+model = timm.create_model("beit_base_patch16_224", pretrained = False)
+model.eval()
+
+# Creating inputs
+input_size = model.default_cfg["input_size"]
+batched_input_size = tuple(batch_size) + input_size
```
Comment (jeremyfowers) [inline models/timm/beit_base_patch16_224.py:15]: This doesn't look right to me, shouldn't it be `batched_input_size = tuple(batch_size) * input_size`?

File: models/timm/beit_base_patch16_224.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# labels: name::beit_base_patch16_224 author::timm
```
Comment (jeremyfowers) [inline models/timm/beit_base_patch16_224.py:1]: Any other labels available? Like the CV task (classification/segmentations/detection)?

Comment (jeremyfowers) [reply models/timm/beit_base_patch16_224.py:1]: Or at least putting "CV" on all them so they can be filtered vs. NLP

File: models/timm/beit_base_patch16_384.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# labels: name::beit_base_patch16_384 author::timm
```
Comment (jeremyfowers) [inline models/timm/beit_base_patch16_384.py:1]: Is the only difference between this model and `beit_base_patch16_224` that this one uses image size 384x384 instead of 224x224?

If so, is it possible to move the image size selection to a `mlagility.parse()` command since there's potentially only ~100 unique models in here not 800+? Or is the image size a property of the model itself?

File: models/timm/efficientnet_b1.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# labels: name::efficientnet_b1 author::timm
```
Comment (jeremyfowers) [inline models/timm/efficientnet_b1.py:1]: Is this going to have a name collision with `models/torch_hub/efficientnet_b1.py`? What happens in the report if I benchmark both `torch_hub` and `timm` corpus?

File: models/timm/skip/convnext_nano_hnf.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
+# labels: name::convnext_nano_hnf author::timm
```
Comment (jeremyfowers) [inline models/timm/skip/convnext_nano_hnf.py:1]: Can you put the skip reason in a comment on each skipped script?

**Review by jeremyfowers — CHANGES_REQUESTED** [review]
Tremendous contribution! Can we discuss the threads I opened before it merges though?

File: README.md [hunk 6]
```diff
@@ -15,13 +15,14 @@ Our _benchit_ CLI allows you to benchmark Pytorch models without changing a sing
 
 <img src="https://github.com/groq/mlagility/raw/main/docs/img/mlagility.gif"  width="800"/>
 
-## Tons of Models
+## 1000+ Models
```
Comment (ramkrishna2910) [inline README.md:18]: Some of these models are duplicates from the other libraries we have. Do we want to remove the duplicates?

File: models/timm/beit_base_patch16_224.py (hunk 1 above)
Comment (danielholanda) [reply models/timm/beit_base_patch16_224.py:15]: I see your confusion. We are simply appending one dimension to the `input_size`. Here is one example from one of the models:

```
>>> batch_size=[1]
>>> input_size=(3,244,244)
>>> tuple(batch_size) + input_size
(1, 3, 244, 244)
```

File: models/timm/beit_base_patch16_384.py (hunk 3 above)
Comment (danielholanda) [reply models/timm/beit_base_patch16_384.py:1]: > Is the only difference between this model and `beit_base_patch16_224` that this one uses image size 384x384 instead of 224x224?

Apart from the input size, the weights of those models are also **should**  be different if we set `pretrained = True`. Therefore, merging models might become problematic if we ever want to use the models with their weights to check for accuracy.

We now have exactly one file for each model in `timm.list_models()`. Some of those are the same model with a different `default_cfg`. Merging models should be possible if we don't care about keeping that 1-1 ratio between `timm.list_models()` and our python files (and if we don't care about the weights).

Comment (jeremyfowers) [reply models/timm/beit_base_patch16_384.py:1]: People keep asking about accuracy tests... so let's take the position that we do care about the weights. Thanks for the explanation, resolving!

File: README.md (hunk 6 above)
Comment (jeremyfowers) [reply README.md:18]: yeah related to https://github.com/groq/mlagility/pull/217#discussion_r1145373659... wondering if its more than just a confusion and might be a bug

File: models/timm/efficientnet_b1.py (hunk 4 above)
Comment (danielholanda) [reply models/timm/efficientnet_b1.py:1]: Great catch. You are right that we now have some overlaps. More precisely, the following models are overlapping:

```
torch_hub/resnet34.py
torch_hub/vgg19_bn.py
torch_hub/densenet169.py
torch_hub/resnext50_32x4d.py
torch_hub/efficientnet_b3.py
torch_hub/efficientnet_b4.py
torch_hub/wide_resnet101_2.py
torch_hub/inception_v3.py
torch_hub/vgg19.py
torch_hub/densenet121.py
torch_hub/densenet161.py
torch_hub/convnext_large.py
torch_hub/vgg11.py
torch_hub/vgg13.py
torch_hub/efficientnet_b0.py
torch_hub/convnext_tiny.py
torch_hub/efficientnet_b2.py
torch_hub/efficientnet_b5.py
torch_hub/vgg16.py
torch_hub/resnet101.py
torch_hub/resnet152.py
torch_hub/efficientnet_b1.py
torch_hub/convnext_small.py
torch_hub/resnet18.py
torch_hub/efficientnet_b6.py
torch_hub/efficientnet_b7.py
torch_hub/convnext_base.py
torch_hub/vgg13_bn.py
torch_hub/vgg11_bn.py
torch_hub/vgg16_bn.py
torch_hub/resnet50.py
torch_hub/wide_resnet50_2.py
torch_hub/densenet201.py
torch_hub/resnext101_32x8d.py
```

What would be your recommendation here?

File: models/timm/beit_base_patch16_224.py (hunk 2 above)
Comment (danielholanda) [reply models/timm/beit_base_patch16_224.py:1]: Added CV as the task of those models

File: models/timm/skip/convnext_nano_hnf.py (hunk 5 above)
Comment (danielholanda) [reply models/timm/skip/convnext_nano_hnf.py]: Added a reason for each of the models! 
Turns out we only have two issues there:

* benchit analysis fails with some pruned models https://github.com/groq/mlagility/issues/219
* Some models use a standard library called InplaceABN and it is not clear whether or not it should become part of our requirements.

File: models/timm/efficientnet_b1.py (hunk 4 above)
Comment (ramkrishna2910) [reply models/timm/efficientnet_b1.py]: `modelname_library` is an option. (At the risk of verbose file names)

Comment (jeremyfowers) [reply models/timm/efficientnet_b1.py]: For all the models in all corpora or just the duplicates?

Comment (jeremyfowers) [reply models/timm/efficientnet_b1.py]: We could leave the script file names alone but change the build names to be like reset34_timms_hash

File: models/timm/skip/convnext_nano_hnf.py (hunk 5 above)
Comment (jeremyfowers) [reply models/timm/skip/convnext_nano_hnf.py]: Why not make InplaceABN part of the requirements?

File: models/timm/beit_base_patch16_224.py (hunk 1 above)
Comment (jeremyfowers) [reply models/timm/beit_base_patch16_224.py:15]: WTF what a weird property of python 😂 

Is there some tuple.append() or something that makes it more obvious what is going on? Or do I just need to git gud?

File: models/timm/efficientnet_b1.py (hunk 4 above)
Comment (ramkrishna2910) [reply models/timm/efficientnet_b1.py]: Yeah thats a better idea. Just edit the build names to include the library

File: models/timm/beit_base_patch16_224.py (hunk 1 above)
Comment (ramkrishna2910) [reply models/timm/beit_base_patch16_224.py:15]: This is weird. Tuples are immutable.

Comment (danielholanda) [reply models/timm/beit_base_patch16_224.py:15]: yeap. That is just how Python works. There's no append() or extend() method for tuples. The + operator seems to be the recommended way of concatenating tuples.

File: models/timm/skip/convnext_nano_hnf.py (hunk 5 above)
Comment (danielholanda) [reply models/timm/skip/convnext_nano_hnf.py]: Interestingly, `InplaceABN` is not a dependency of `timm`.

`InplaceABN` requires `CUDA` versions >= 10.0, which is a **big** dependency. I would probably recommend against having `InplaceABN` as part of our `requirements.txt` file to avoid the `CUDA` dependency.

**Review by ramkrishna2910 — APPROVED** [review]

File: models/timm/skip/convnext_nano_hnf.py (hunk 5 above)
Comment (jeremyfowers) [reply models/timm/skip/convnext_nano_hnf.py]: Can you run `conda list` in your environment? I get:
```
nvidia-cublas-cu11        11.10.3.66               pypi_0    pypi
nvidia-cuda-cupti-cu11    11.7.101                 pypi_0    pypi
nvidia-cuda-nvrtc-cu11    11.7.99                  pypi_0    pypi
nvidia-cuda-runtime-cu11  11.7.99                  pypi_0    pypi
nvidia-cudnn-cu11         8.5.0.96                 pypi_0    pypi
nvidia-cufft-cu11         10.9.0.58                pypi_0    pypi
nvidia-curand-cu11        10.2.10.91               pypi_0    pypi
nvidia-cusolver-cu11      11.4.0.1                 pypi_0    pypi
nvidia-cusparse-cu11      11.7.4.91                pypi_0    pypi
nvidia-nccl-cu11          2.14.3                   pypi_0    pypi
nvidia-nvtx-cu11          11.7.91                  pypi_0    pypi
```

So I think we already have the CUDA dep?

**Review by jeremyfowers — APPROVED** [review]
Approving but the InplaceABN discussion doesnt feel resolved yet
